Remove derivative of specific heat from NS kernels

GiudGiud opened the pull request:
If h = cp T with cp constant it doesnt matter
For our future formulation of h = int(cp dT) then it ll matter

Refs #21245

GiudGiud committed:
Remove specific heat derivatives from NS kernels, refs #21245

GiudGiud committed:
Remove from NS tests, refs #21245

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1111927/) on ec1ded4 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/21593/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1112365/) on 26d1f10 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>fcc670</th>
      <th colspan=4>#21593 26d1f1</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>83.55%</td>
      <td>83.55%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>77926</td>
      <td>77925</td>
      <td>-1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15346</td>
      <td>15347</td>
      <td>+1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/21593/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/21593/coverage/framework)


## Modules coverage

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>fcc670</th>
      <th colspan=4>#21593 26d1f1</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.43%</td>
      <td>85.44%</td>
      <td>+0.00%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>9690</td>
      <td>9679</td>
      <td>-11</td>
      <td>2</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1652</td>
      <td>1650</td>
      <td>-2</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/21593/coverage_diff/navier_stokes)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/21593/coverage/navier_stokes)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/21593/coverage/xfem)

</details>



This comment will be updated on new commits.

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/PINSFVEnergyTimeDerivative.md:
```diff
@@ -13,8 +13,7 @@ for the fluid phase and
 
-The time derivatives of density and the specific heat capacity are ignored if [!param](/FVKernels/PINSFVEnergyTimeDerivative/drho_dt)
-and [!param](/FVKernels/PINSFVEnergyTimeDerivative/dcp_dt) are not provided. For incompressible flows, the former should not be provided.
+The time derivative of the density is ignored if [!param](/FVKernels/PINSFVEnergyTimeDerivative/drho_dt) is not provided. For incompressible flows, the former should not be provided.
```
I think we should include the derivation of the correct form in the description before this. I can append it to this PR if needed.

GiudGiud replied:
added!

GiudGiud committed:
Add derivation for the energy time derivative documentation, refs 21245

Update modules/navier_stokes/doc/content/source/fvkernels/INSFVEnergyTimeDerivative.md

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

Update modules/navier_stokes/doc/content/source/fvkernels/INSFVEnergyTimeDerivative.md

Co-authored-by: Peter German <31662443+grmnptr@users.noreply.github.com>

GiudGiud commented:
We should probably get things in order for enthalpy vs energy too at some point.

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVEnergyTimeDerivative.md:
```diff
@@ -5,14 +5,34 @@
+we take the partial derivative with regards to time:
+
+\begin{equation}
+\frac{\partial U}{\partial t}} = \frac{\partial \rho(T, p)}{\partial t}} \int_0^T dT cp(T, p) + \rho(T, p) \frac{\partial T}{\partial t}} cp(T, p)
```
This does not parse well.
```suggestion
\frac{\partial U}{\partial t}} = \frac{\partial \rho(T, p)}{\partial t} \int_0^T dT cp(T, p) + \rho(T, p) \frac{\partial T}{\partial t} cp(T, p)
```

grmnptr reviewed (changes requested):
The equation in the derivation doesn't compile.

GiudGiud commented:
should be good now

grmnptr commented on modules/navier_stokes/doc/content/source/fvkernels/INSFVEnergyTimeDerivative.md:
```diff
@@ -5,14 +5,34 @@
+we take the partial derivative with regards to time:
+
+\begin{equation}
+\frac{\partial U}{\partial t}} = \frac{\partial \rho(T, p)}{\partial t} \int_0^T dT cp(T, p) + \rho(T, p) \frac{\partial T}{\partial t} cp(T, p)
```
One more, maybe you can squash this guy.
```suggestion
\frac{\partial U}{\partial t} = \frac{\partial \rho(T, p)}{\partial t} \int_0^T dT cp(T, p) + \rho(T, p) \frac{\partial T}{\partial t} cp(T, p)
```

GiudGiud commented:
done. I would build docs locally but all CPUs are on NONUNITY build
